### PR TITLE
Add mapbox-gl via single package.json w/ npm auto-update

### PR DESCRIPTION
**Pull request by jasonbarry**
Pull request for issue: #10654 (supersedes PR #10734, #10745, #10746, #10767)

# Profile of the lib
 * Git repository (required): https://github.com/mapbox/mapbox-gl-js
 * Official website (optional, not the repository): https://www.mapbox.com/mapbox-gl-js/api/
 * NPM package url (optional): https://www.npmjs.com/package/mapbox-gl
 * License and its reference: [BSD-3-Clause](https://www.npmjs.com/package/mapbox-gl)
 * GitHub / Bitbucket popularity (required): 
   - Count of watchers: 221
   - Count of stars: 1554
   - Count of forks: 367
 * NPM download stats (optional):
   - Downloads in the last day: 962
   - Downloads in the last week: 11352
   - Downloads in the last month: 46516

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/mapbox-gl/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "mapbox-gl",
+  "description": "A WebGL interactive maps library",
+  "filename": "mapbox-gl.js",
+  "license": "BSD-3-Clause",
+  "homepage": "https://www.mapbox.com/mapbox-gl-js/api/",
+  "author": "Konstantin Kaefer (mail@kkaefer.com)",
```
Comment (pvnr0082t) [inline ajax/libs/mapbox-gl/package.json]: How do you make sure he is the author or maintainer? Any reference? Thanks.

Comment (jasonbarry) [inline ajax/libs/mapbox-gl/package.json]: The first commit of the repo is by him, and his GH profile says he still works at Mapbox. https://github.com/kkaefer

**Comment by jasonbarry** [comment]
@pvnr0082t I opened https://github.com/mapbox/mapbox-gl-js/issues/4370

**Comment by PeterDaveHello** [comment]
@jasonbarry thanks

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by jasonbarry** [comment]
@PeterDaveHello I haven't heard anything back from the mapbox team. Is the minified CSS required to host the minified JS on CDNJS? Would it be possible to host the JS only without depending on the CSS to be minified?

**Comment by PeterDaveHello** [comment]
Should be required, I think the project is still alive and very active, can you just ping its author for help in that issue? Thanks.

**Comment by jasonbarry** [comment]
@PeterDaveHello I've submitted a PR. https://github.com/mapbox/mapbox-gl-js/pull/4493

**Comment by PeterDaveHello** [comment]
hmmm... seem to be rejected...

**Comment by jasonbarry** [comment]
@PeterDaveHello from the rejected PR: 

> The benefits of pre-compressing CSS are much reduced when taking into account gzip content encoding. In my measurements, it goes from 7257 bytes to 7094 bytes. I don't think a 163 byte improvement is enough to justify the additional complexity in the build pipeline

What next?

**Comment by PeterDaveHello** [comment]
As we don't support partially minify process ... let me think about it ...

**Comment by PeterDaveHello** [comment]
@pvnr0082t would you please help make sure everything is good first? Thanks!

**Comment by pvnr0082t** [comment]
@PeterDaveHello Everything else looks fine except partially minify problem. :+1:

**Comment by maruilian11** [comment]
@jasonbarry Since git repo has more stable versions, could you please turn npm auto-update to the git one. Thanks!

**Comment by maruilian11** [comment]
sorry, just found js files are ignored on git repo.

File: ajax/libs/mapbox-gl/package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "mapbox-gl",
+  "description": "A WebGL interactive maps library",
+  "filename": "mapbox-gl.js",
+  "license": "BSD-3-Clause",
+  "homepage": "https://www.mapbox.com/mapbox-gl-js/api/",
```
Comment (maruilian11) [inline ajax/libs/mapbox-gl/package.json]: This page get 404 error.
Maybe we should turn to use https://www.mapbox.com/maps/?

**Comment by maruilian11** [comment]
@cdnjs/intern2 we may need to add previous versions(before 0.3.1) from git repo manually for this lib.

File: ajax/libs/mapbox-gl/package.json (hunk 2 above)
Comment (jasonbarry) [inline ajax/libs/mapbox-gl/package.json]: Try again? I'm able to load https://www.mapbox.com/mapbox-gl-js/api/. Thanks!

Comment (PeterDaveHello) [inline ajax/libs/mapbox-gl/package.json]: https://www.mapbox.com/mapbox-gl-js/api/ looks good to me

**Comment by PeterDaveHello** [comment]
we can take care of old versions later

**Comment by PeterDaveHello** [comment]
@pvnr0082t looks like you didn't **approve** it?

**Comment by pvnr0082t** [comment]
@PeterDaveHello Approved. Thanks.

**Comment by PeterDaveHello** [comment]
@jasonbarry thanks for your contribution!

**Pull request merged**